### ISSUE TITLE
azurerm_automation_runbook doesn't create python runbook.

### ISSUE THREAD

**Issue by mmclane** [hashicorp/terraform-provider-azurerm#1602]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version
Terraform v0.11.7
+ provider.azurerm v1.8.0
+ provider.template v1.0.0

### Affected Resource(s)
* azurerm_automation_runbook

### Terraform Configuration Files

```hcl
resource "azurerm_automation_runbook" "runbook" {
  name                = "test-kitchen-cleanup"
  location            = "${azurerm_resource_group.autorg.location}"
  resource_group_name = "${azurerm_resource_group.autorg.name}"
  account_name        = "${azurerm_automation_account.account.name}"
  log_verbose         = "true"
  log_progress        = "true"
  description         = "Hello World script, just to get things created"
  runbook_type        = "Script"

  publish_content_link {
    uri = "${azurerm_storage_blob.script.url}"
  }
}
```

### Expected Behavior
* Creates a runbook in the specified automation account
* The runbook created is of type Python 2 Runbook

### Actual Behavior
* Creates the runbook and reports success.
* The resulting runbook is actually a PowerShell Workflow Runbook.
* TF state file show the following:

```hcl
azurerm_automation_runbook.runbook:
  id = /subscriptions/8efe2bb2-0b97-4186-8c1e-c75e6c772a98/resourceGroups/test-kitchen-cleanup-preprod/providers/Microsoft.Automation/automationAccounts/test-kitchen-cleanup-preprod/runbooks/test-kitchen-cleanup
  account_name = test-kitchen-cleanup-preprod
  description = Hello World script, just to get things created
  location = eastus2
  log_progress = true
  log_verbose = true
  name = test-kitchen-cleanup
  publish_content_link.# = 1
  publish_content_link.0.hash.# = 0
  publish_content_link.0.uri = https://crtoutilityusepreprod.blob.core.windows.net/scripts/test-kitchen-cleanup.ps1
  publish_content_link.0.version =
  resource_group_name = test-kitchen-cleanup-preprod
  runbook_type = Script
  tags.% = 0
```
### Important Factoids
The documentation is vague on if TF supports creating python based runbooks.  It says the following:
runbook_type - (Required) The type of the runbook - can be either Graph, GraphPowerShell, GraphPowerShellWorkflow, PowerShellWorkflow, PowerShell or Script.

I would assume that Script is a python runbook since the valid types of runbooks are:
* Powershell
* Python 2 (preview)
* Graphical
* PowerShell Workflow
* Graphical PowerShell Workflow

**Comment by mmclane** [comment]
I just noticed that it shows the content link being a ps1 file.  I can confirm that it doesn't matter if its a py file, it still doesn't set the runbook type correctly.

**Comment by tombuildsstuff** [comment]
Upstream issue: 
https://github.com/Azure/azure-sdk-for-go/issues/2256

**Comment by hdost** [comment]
For reference, sounds like the problem might start with Azure/azure-rest-api-specs/issues#3457

**Comment by mmclane** [comment]
If this is going to be an enhancement that relies on upstream work, I suggest that you clarify in the documentation what the script type is and that python runbooks are not yet supported.

**Comment by bowlofstew** [comment]
Bump on this issue.  I'd love to use this feature.  If it isn't coming (last update was 7/20/2018), can you update the documentation please?

**Comment by ffais** [comment]
Any news ?

**Comment by JukkaUusisalo** [comment]
Anything?

**Comment by michaelmcmillan** [comment]
Does anyone have an idea of where it breaks down? We want to use Python for our Runbooks, but it doesn't seem like it's possible atm.

**Comment by jcbagtas** [comment]
I can't believe I hit a terraform limit that is 2 years old. lol
Is there any news about this?

**Comment by colinuipath** [comment]
It has been 2 years and Azure is just not doing its job!

**Comment by JamesStanley** [comment]
If you use the API directly (even the 2015 version) specifying the runbook type as "Python3" works..........![image](https://user-images.githubusercontent.com/42772711/110143159-d5490700-7dce-11eb-8392-ac1c4a88fc91.png)

(Anyone testing will need to update the URI to something valid for it to succeed.)

However even the latest version of the API still doesn't list Python as an option...https://github.com/Azure/azure-rest-api-specs/blob/782bab9c34014092f48de95874a71e7ecd1a9bba/specification/automation/resource-manager/Microsoft.Automation/stable/2019-06-01/runbook.json#L1154

**Comment by KoenR3** [comment]
Also having this issue

**Comment by lumy** [comment]
Any news on this ? from what i understood it is possible if you hit directly the api, will terraform manage this sooner or later ?

**Comment by corticalstack** [comment]
Any update on AzureRM provider support for Python runbooks? Thanks

**Comment by chnusti** [comment]
I reached out Microsoft support regarding this issue and tried to explain the discrepancy between implementation and documentation.
the engineer was pretty much sure the api supports type python3 even though not written in the docs.
moreover I also provide him this github issue regarding this problem.

They will get back to me soon.

**Comment by wjg659** [comment]
It looks like this is a Microsoft issue.  While the API may accept Python3, their enumeration of acceptable runbook types does not have Python or Python3 in it. 

```
class RunbookTypeEnum(with_metaclass(_CaseInsensitiveEnumMeta, str, Enum)):
    """Gets or sets the type of the runbook.
    """

    SCRIPT = "Script"
    GRAPH = "Graph"
    POWER_SHELL_WORKFLOW = "PowerShellWorkflow"
    POWER_SHELL = "PowerShell"
    GRAPH_POWER_SHELL_WORKFLOW = "GraphPowerShellWorkflow"
    GRAPH_POWER_SHELL = "GraphPowerShell"
```


Above code taken from [here]( https://github.com/Azure/azure-sdk-for-python/blob/c8291ac6cb0dbd865da03a88dd2bcb9279e2c4a6/sdk/automation/azure-mgmt-automation/azure/mgmt/automation/models/_automation_client_enums.py#L227).

The terraform code preventing us from trying to pass Pyton3 in appears to be this block:

```
				ValidateFunc: validation.StringInSlice([]string{
					string(automation.RunbookTypeEnumGraph),
					string(automation.RunbookTypeEnumGraphPowerShell),
					string(automation.RunbookTypeEnumGraphPowerShellWorkflow),
					string(automation.RunbookTypeEnumPowerShell),
					string(automation.RunbookTypeEnumPowerShellWorkflow),
					string(automation.RunbookTypeEnumScript),
				}, !features.ThreePointOhBeta()),
```

Found from [here](https://github.com/hashicorp/terraform-provider-azurerm/blob/47f4d08ca05d86d89c361ffd7bb623ae684df292/internal/services/automation/automation_runbook_resource.go#L70).

The minimum fix for this appears to be Microsoft adding 

```
PYTHON = "Python"
PYTHON3 = "Python3"
```

...to their enumeration block and someone on the Terraform side adding the following to the validation block:
```
string(automation.RunbookTypePython),
string(automation.RunbookTypePython3)
```

**Comment by DuneganS** [comment]
Hello,

Is there any type of update on this issue from any source? It still seems to be an issue with azurerm 3.19.1

I've zipped and attached the basic testing that I did. Running

`terraform init`
`terraform validate`

Gives this result

```
PS E:\NetShare\_coding\terraform\azure> terraform.exe init

Initializing the backend...

Initializing provider plugins...
- Finding latest version of hashicorp/local...
- Finding hashicorp/azurerm versions matching "~> 3.19.1"...
- Installing hashicorp/local v2.2.3...
- Installed hashicorp/local v2.2.3 (signed by HashiCorp)
- Installing hashicorp/azurerm v3.19.1...
- Installed hashicorp/azurerm v3.19.1 (signed by HashiCorp)

Terraform has created a lock file .terraform.lock.hcl to record the provider
selections it made above. Include this file in your version control repository
so that Terraform can guarantee to make the same selections by default when
you run "terraform init" in the future.

Terraform has been successfully initialized!

You may now begin working with Terraform. Try running "terraform plan" to see
any changes that are required for your infrastructure. All Terraform commands
should now work.

If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your working directory. If you forget, other
commands will detect it and remind you to do so if necessary.
PS E:\NetShare\_coding\terraform\azure> terraform.exe validate
╷
│ Error: expected runbook_type to be one of [Graph GraphPowerShell GraphPowerShellWorkflow PowerShell PowerShellWorkflow Script], got Python3
│ 
│   with azurerm_automation_runbook.azure_automation_runbook_001,
│   on main.tf line 37, in resource "azurerm_automation_runbook" "azure_automation_runbook_001":
│   37:   runbook_type            = "Python3"
```

[gh-1602.zip](https://github.com/hashicorp/terraform-provider-azurerm/files/9391077/gh-1602.zip)

**Comment by catriona-m** [comment]
It looks like this has now been resolved by #18921 so this can be closed now. Thanks!
